drv/Lists.drv

```diff
@@ -47,6 +47,26 @@ public final class LISTS {
 		return l;
 	}
 
+	public static KEY_GENERIC LIST KEY_GENERIC reverse(final LIST KEY_GENERIC l) {
+		int size = l.size();
+		if(l instanceof RandomAccess) {
+			for (int i=0, mid=size>>1, j=size-1; i<mid; i++, j--) {
+				final KEY_GENERIC_TYPE t = l.GET_KEY(i);
+				l.set(i, l.GET_KEY(j));
+				l.set(j, t);
+			}
+			return l;
+		}
+		KEY_LIST_ITERATOR KEY_GENERIC fwd = l.listIterator();
+		KEY_LIST_ITERATOR KEY_GENERIC rev = l.listIterator(size);
+		for (int i=0, mid=size>>1, j=size-1; i<mid; i++, j--) {
```

Comment:
I think this should be `size >>> 1` (not `size >> 1`). However this won't have any practical impact.

Also: Formatting. Should have some more spaces :-)

Reply:
To be fair this is about negative values. If you have those you have other problems.
Thanks.

Reply:
Oh wait nevermind, its not relevant here, only for binary search.